interactions/ext/hybrid_commands/hybrid_slash.py

```diff
@@ -0,0 +1,593 @@
+import asyncio
+import inspect
+from typing import Any, Callable, List, Optional, Union, TYPE_CHECKING, Awaitable
+
+import attrs
+from interactions import (
+    BaseContext,
+    Converter,
+    NoArgumentConverter,
+    Attachment,
+    SlashCommandChoice,
+    OptionType,
+    BaseChannelConverter,
+    ChannelType,
+    BaseChannel,
+    BaseCommand,
+    MemberConverter,
+    UserConverter,
+    RoleConverter,
+    SlashCommand,
+    SlashContext,
+    Absent,
+    LocalisedName,
+    LocalisedDesc,
+    MISSING,
+    SlashCommandOption,
+    Snowflake_Type,
+    Permissions,
+)
+from interactions.client.const import AsyncCallable, GLOBAL_SCOPE
+from interactions.client.utils.serializer import no_export_meta
+from interactions.client.utils.misc_utils import maybe_coroutine, get_object_name
+from interactions.client.errors import BadArgument
+from interactions.ext.prefixed_commands import PrefixedCommand, PrefixedContext
+from interactions.models.internal.converters import _LiteralConverter
+from interactions.models.internal.checks import guild_only
+
+if TYPE_CHECKING:
+    from .context import HybridContext
+
+__all__ = ("HybridSlashCommand", "hybrid_slash_command", "hybrid_slash_subcommand")
+
+
+def _values_wrapper(a_dict: dict | None) -> list:
+    return list(a_dict.values()) if a_dict else []
+
+
+def generate_permission_check(permissions: "Permissions") -> Callable[["HybridContext"], Awaitable[bool]]:
+    async def _permission_check(ctx: "HybridContext") -> bool:
+        return ctx.author.has_permission(*permissions) if ctx.guild_id else True  # type: ignore
+
+    return _permission_check  # type: ignore
+
+
+def generate_scope_check(_scopes: list["Snowflake_Type"]) -> Callable[["HybridContext"], Awaitable[bool]]:
+    scopes = frozenset(int(s) for s in _scopes)
+
+    async def _scope_check(ctx: "HybridContext") -> bool:
+        return int(ctx.guild_id) in scopes
+
+    return _scope_check  # type: ignore
+
+
+class BasicConverter(Converter):
+    def __init__(self, type_to_convert: Any) -> None:
+        self.type_to_convert = type_to_convert
+
+    async def convert(self, ctx: BaseContext, arg: str) -> Any:
+        return self.type_to_convert(arg)
+
+
+class BoolConverter(Converter):
+    async def convert(self, ctx: BaseContext, argument: str) -> bool:
+        lowered = argument.lower()
+        if lowered in {"yes", "y", "true", "t", "1", "enable", "on"}:
```

Comment:
Maybe we should take these literal sets and move them to variable declarations so we or developers can modify them

Reply:
Like the idea, but as of right now, that converter's meant to have parity with prefixed commands. I'll add it in another PR for prefixed commands though - I do like it.